Index: b/ffmpeg-2-8-7/configure
===================================================================
--- a/ffmpeg-2-8-7/configure
+++ b/ffmpeg-2-8-7/configure
@@ -3577,6 +3577,7 @@ probe_cc(){
         fi
         _cflags_speed='-O3'
         _cflags_size='-Os'
+        case "$(uname -m)" in s390*) _cflags_speed='-O2'; esac
     elif $_cc --version 2>/dev/null | grep -q ^icc; then
         _type=icc
         _ident=$($_cc --version | head -n1)
